resources/metric-ui.mjs

```diff
@@ -0,0 +1,221 @@
+                    <td>${metric.unit}</td>
+                </tr>`
+        )
+        .join("");
+    const subMetrics = renderChildren ? renderSubMetrics(params) : "";
+    return `
+        <dl class="metric">
+            <dt><h3>${title}<h3></dt>
+            <dd>
+                <div class="metric-chart">
+                    ${scatterPlot}
+                    <table class="chart chart-legend">${legend}</table>
+                </div>
+                ${subMetrics}
+            </dd>
+        </dl>
+    `;
+}
+
+function renderSubMetrics(params) {
+    let { metric, width, subMetricMargin, colors = COLORS } = params;
+    const children = metric.children;
+    const hasChildMetric = children.length > 0 && children[0].children.length > 0;
+    if (!hasChildMetric)
+        return "";
+    const subMetricWidth = width - subMetricMargin;
+    let childColors = [...colors];
+    const subMetrics = metric.children
+        .map((metric) => {
+            // Rotate colors to get different colors for sub-plots.
+            for (let i = 0; i < metric.children.length; i++) {
+                const color = childColors.pop();
+                childColors.unshift(color);
+            }
+            const subMetricParams = { ...params, metric, title: metric.name, width: subMetricWidth, colors: childColors };
+            return renderMetricView(subMetricParams);
+        })
+        .join("");
+    return `
+        <label class="details-toggle">
+            <input type="checkbox" 
+                    onclick="this.parentNode.nextElementSibling.classList.toggle('visible')" />
+            Details
+        </label>
+        <div class="submetrics">
+            ${subMetrics}
+        </div>
+    `;
+}
+
+function prepareScatterPlotValues(metric, normalize = true) {
+    let points = [];
+    const metrics = metric.children;
+    // Arrange child-metrics values in a single coordinate system:
+    // - metric 1: x values are in range [0, 1]
+    // - metric 2: y values are in range [1, 2]
+    // - ...
+    // This way each metric data point is on a separate track in the scatter
+    // plot.
+    // All x values are normalized by the mean of each metric and centered on 0.
+    // Example: [90ms, 100ms, 110ms] =>  [-10%, 0%, +10%]
+    const toPercent = 100;
+    for (let metricIndex = 0; metricIndex < metrics.length; metricIndex++) {
+        const subMetric = metrics[metricIndex];
+        // Add coordinated for deviation rect:
+        const mean = subMetric.mean;
+        const unit = subMetric.unit;
+        let width = subMetric.delta;
+        let center = mean;
+        if (normalize) {
+            width = (subMetric.delta / mean) * toPercent;
+            center = 0;
+        }
+        let left = center - width / 2;
+        const y = metricIndex;
+        const label = `Mean: ${subMetric.valueString}\n` + `Min: ${subMetric.min.toFixed(2)}${unit}\n` + `Max: ${subMetric.max.toFixed(2)}${unit}`;
```

Comment:
nit: Since we're using template literals, the plus sign can get removed I believe.